compiler/rustc_typeck/src/errors.rs

```diff
@@ -270,7 +271,7 @@ impl<'a> SessionDiagnostic<'a> for MissingTypeParams {
 
         let mut suggested = false;
         if let (Ok(snippet), true) = (
-            sess.source_map().span_to_snippet(self.span),
+            self.source_map.span_to_snippet(self.span),
```

Comment:
Same with precomputing the `span_to_snippet` here

Reply:
Addressed here https://github.com/rust-lang/rust/pull/101434/commits/dd5850b8faf04e1b98542cf0813920385a6cc4db.
Thank you!